### ISSUE TITLE
pnpm dlx cannot fetch versions for library "instantsearch.js"

### ISSUE THREAD

**Issue by ghost** [pnpm/pnpm#7418]
### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

8.12.1

### Which area(s) of pnpm are affected? (leave empty if unsure)

Package manager compatibility

### Reproduction steps

I was trying to use `pnpm dlx` to fetch a package from the registry and create an instansearch app. The following error occurred: 
```
Cannot fetch versions for library "instantsearch.js".
```
I wonder if this is a proxy issue. Maybe a verbose or debug mode of pnpm would help resolve, and I wonder how to enable such mode using pnpm.

### Describe the Bug

Cannot fetch versions for library "instantsearch.js".

### Expected Behavior

pnpm would list the available versions of InstantSearch.js so I can choose the desired version number. Now only 1.0.0 displayed. 

### Which Node.js version are you using?

21.4.0

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by KSXGitHub** [comment]
I ran `pnpm dlx instantsearch.js` and the error says "no binary". What command did you use?

**Comment by gustavoquincy** [comment]
You can use `pnpm dlx create-instantsearch-app typesense`.

**Comment by KSXGitHub** [comment]
@gustavoquincy I cannot reproduce the error. It asks me a bunch of question then execute `yarn`.

**Comment by gustavoquincy** [comment]
Right, when I switched my internet environment the other day, the error disappeared as well. Pretty sure the registry of yarn matters here:
```
yarn config list
yarn config v1.22.21
info yarn config
{
  'version-tag-prefix': 'v',
  'version-git-tag': true,
  'version-commit-hooks': true,
  'version-git-sign': false,
  'version-git-message': 'v%s',
  'init-version': '1.0.0',
  'init-license': 'MIT',
  'save-prefix': '^',
  'bin-links': true,
  'ignore-scripts': false,
  'ignore-optional': false,
  registry: 'https://registry.yarnpkg.com',
  'strict-ssl': true,
  'user-agent': 'yarn/1.22.21 npm/? node/v21.4.0 darwin arm64',
  lastUpdateCheck: 1702694946577
}
info npm config
{
  registry: 'http://registry.npmjs.org/'
}
✨  Done in 0.02s.
```

**Comment by KSXGitHub** [comment]
So was this a yarn error or pnpm error?

**Comment by gustavoquincy** [comment]
pnpm calls yarn right? Or it use the configurations in ~/.npmrc?

**Comment by KSXGitHub** [comment]
pnpm does not call yarn, `create-instantsearch-app` calls yarn.

**Comment by gustavoquincy** [comment]
I see.

**Comment by zkochan** [comment]
Closing for lack of a reproduction on a supported version.

The version list could not be fetched from the registry, which points at the network or a proxy. Please open a new issue if a current pnpm still fails; `--loglevel debug` prints the request it makes.

If you still hit this on the latest pnpm, please comment with a reproduction and we will reopen.

---
Written by an agent (Claude Code, claude-opus-5).
